File: ts-sdk/src/sdk/models/shared/folderlocks.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { FolderLock } from "./folderlock";



// FolderLocks
/** 
 * The part of an API response that describes marker
 * based pagination
**/
export class FolderLocks extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=entries", elemType: FolderLock })
  entries?: FolderLock[];

  @SpeakeasyMetadata({ data: "json, name=limit" })
  limit?: number;

  @SpeakeasyMetadata({ data: "json, name=next_marker" })
  nextMarker?: number;

  @SpeakeasyMetadata({ data: "json, name=prev_marker" })
  prevMarker?: number;
}

File: java-sdk/lib/src/main/java/models/shared/FolderLocks.java
package .models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * FolderLocks
 * The part of an API response that describes marker
 * based pagination
**/
public class FolderLocks {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("entries")
    public FolderLock[] entries;
    public FolderLocks withEntries(FolderLock[] entries) {
        this.entries = entries;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("limit")
    public Long limit;
    public FolderLocks withLimit(Long limit) {
        this.limit = limit;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("next_marker")
    public Long nextMarker;
    public FolderLocks withNextMarker(Long nextMarker) {
        this.nextMarker = nextMarker;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("prev_marker")
    public Long prevMarker;
    public FolderLocks withPrevMarker(Long prevMarker) {
        this.prevMarker = prevMarker;
        return this;
    }
}